src/pubtools/sign/ansible/clear_sign.py

```diff
@@ -0,0 +1,151 @@
+#!/usr/bin/python
+
+from __future__ import absolute_import, division, print_function
+
+from ansible.module_utils.basic import AnsibleModule
+
+from ..operations.clearsign import ClearSignOperation
+from ..signers.msgsigner import _msg_clear_sign
+
+__metaclass__ = type
+
+doc_arguments = ClearSignOperation.doc_arguments()
+
+DOCUMENTATION = r"""
+---
+module: clear_sign
+version_added: "0.1"
+short_description: This is used to do clear sign
+description:
+    - Sign data with clear sign.
+options:
```

Comment:
I didn't have a chance to take a look a this properly when I was writing code modules and I hoped I can work on this task later. But since you took it already, my idea was to have this completely dynamically generated. To do that, Signer.doc_arguments needs to be changed to return mapping {key :  (type, description, required) }. Which should be enough to generate this whole paragraph dynamically. Feel free to extend Signer class with method for example doc_arguments_yaml (or something similar) which can be then used here.

Reply:
Made changes to generate documents, examples and returns dynamically in the new commit. Can you please review it again?